src/track/frame.h

```diff
@@ -0,0 +1,102 @@
+#pragma once
+
+class QDebug;
+
+namespace mixxx {
+class Frame {
+  public:
+    Frame()
+            : m_dFrame(0) {
+    }
+
+    explicit Frame(double dFrame)
+            : m_dFrame(dFrame) {
+    }
+
+    ~Frame() = default;
+
+    void setValue(double dFrame) {
+        m_dFrame = dFrame;
+    }
+
+    double getValue() const {
+        return m_dFrame;
+    }
+
+    Frame& operator+=(const Frame& arg) {
+        m_dFrame += arg.getValue();
+        return *this;
+    }
+
+    Frame& operator-=(const Frame& arg) {
+        m_dFrame -= arg.getValue();
+        return *this;
+    }
+
+    Frame& operator*=(const double arg) {
+        m_dFrame *= arg;
+        return *this;
+    }
+
+    Frame& operator/=(const double arg) {
+        m_dFrame /= arg;
+        return *this;
+    }
+
+    Frame operator++(int) {
+        Frame temp(*this);
+        m_dFrame++;
+        return temp;
+    }
+
+  private:
+    double m_dFrame;
```

Comment:
Please omit the type prefix, this is considered an anti-pattern.

Instead define a public `typedef double value_t` (see RgbColor) and replace this member with `value_t m_val;`. Use `Frame::value_t` instead of `double` whenever you refer to this inner type.